### PR TITLE
Fix elytron properties perf issues

### PR DESCRIPTION
**Pull request by loicmathieu**
During some performance testing, I discover, thanks to profiler CPU flamegraph, that the Elytron Properties implementation uses a lot of CPU cycles to create the `WilldFlyElytronProvider`.

It apears that the security provider supplier is called on each request calls, and triggers the creation of a new `WilldFlyElytronProvider`.

Creating a unique `WilldFlyElytronProvider` and returing it from the supplier instead of creating it each time fix the issue.

### CODE REVIEW

File: extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java [hunk 1]
```diff
@@ -170,9 +170,11 @@ public void run() {
         SecurityRealm realm = LegacyPropertiesSecurityRealm.builder()
                 .setDefaultRealm(config.realmName)
                 .setProviders(new Supplier<Provider[]>() {
+                    Provider[] providers = new Provider[] { new WildFlyElytronProvider() };
+
                     @Override
                     public Provider[] get() {
-                        return new Provider[] { new WildFlyElytronProvider() };
+                        return providers;
```
Comment (gsmet) [inline extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java]: Maybe you can make the whole supplier static and reuse it at both locations?

**Review by gsmet — CHANGES_REQUESTED** [review]
Good catch. I added a comment to further improve the situation.

**Comment by gsmet** [comment]
Also, can you check in the other security implementation that we don't have this pattern?

File: extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java (hunk 1 above)
Comment (loicmathieu) [reply extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java]: done

**Comment by loicmathieu** [comment]
@gsmet I add the same performance optimization to the JDBC implementation.

OAuth2 and OIDC didn't suffers from the same issue.

**Comment by loicmathieu** [comment]
@gsmet I also review the LDAP and JPA security extensions currently under review, LDAP has the same issue (I add a review comment), JPA doesn't seems to have it (not 100% sure as it's a complex extension and I just have a quick look).

**Review by stuartwdouglas — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**